File: assignments/assignment-1-solution/src-csharp/CrazyEights/Domain/IsPlayable.cs
namespace CrazyEights.Domain;

/// <summary>
/// Represents the outcome of a playability check.
/// </summary>
public class IsPlayable
{
    public IsPlayable(bool canPlay, string reason = "")
    {
        CanPlay = canPlay;
        Reason = reason;
    }

    public bool CanPlay { get; }

    public string Reason { get; }
}

File: assignments/assignment-1-solution/src-java/src/main/java/edu/kennesaw/crazy8s/domain/IsPlayable.java
package edu.kennesaw.crazy8s.domain;

/**
 * Represents the outcome of a playability check.
 */
public class IsPlayable {
    private final boolean canPlay;
    private final String reason;

    public IsPlayable(boolean canPlay, String reason) {
        this.canPlay = canPlay;
        this.reason = reason;
    }

    public boolean canPlay() {
        return canPlay;
    }

    public String getReason() {
        return reason;
    }
}
